hs628_tools/feature_missing.py

```diff
@@ -0,0 +1,188 @@
+            -----------
+            Parameters:
+                data: numpy ndarray, shape = [n_samples, n_features]
+                      observations and features
+                corr_thresh: float between 0 and 1
+                      Pre-defined threshold set by the user, default value = 0.9
+            -----------
+            Returns:
+                A list of tuples containing a pair of column indices with
+                correlation coefficient value more than specified value
+        """
+
+        # creating a matix for correlation coefficient
+        corr_matrix = np.zeros((data.shape[1], data.shape[1]))
+
+        for i in range(data.shape[1]):
+            for j in range(data.shape[1]):
+                # Calculating the corrcoef of each pair and save in corr_matrix
+                corr_matrix[i, j] = np.corrcoef(data[:, i], data[:, j])[0, 1]
+        # finding pairs with corrcoef more than corr_thresh and save them as a tuple in corr_pairs
+        corr_pairs = []
+        for i in range(corr_matrix.shape[0]):
+            for j in range(i + 1, corr_matrix.shape[0]):
+                if abs(corr_matrix[i, j]) > corr_thresh:
+                    corr_pairs += [(i, j)]
+
+        print('%d features with a correlation greater than %0.2f:\n\n' % (len(corr_pairs), corr_thresh))
+        for i in corr_pairs:
+            print('columns #%d and #%d have a correlation value of %s \n' % (i[0], i[1], corr_matrix[i[0], i[1]]))
+        self.collinear_col = np.array(corr_pairs)
+        return self.collinear_col
+
+    def low_importance(self, features, target, target_type="classification", importance_thresh=1 / features.shape[1]):
+        # type: (ndarray, ndarray, str, float) -> ndarray
+
+        """
+        Find columns with their normalized importance value below pre-defined threshold using RandomForestRegressor or
+        RandomForestClassifier based on the target_type.
+            -----------
+            Parameters:
+                features :  numpy ndarrays, , shape = [n_samples, n_features]
+                            observations and features
+
+                target:     numpy ndarrays, shape = (1, ) target can be binary if target_type is classification or
+                            continuous if target_type is regression
+
+                target_type : string defines the RandomForest method if target is continuous or binary. default value
+                            = "classification"
+
+                importance_thresh : float between 0 and 1
+                            Pre-defined threshold set by the user, default value = 1/number of the features
+            -----------
+            Returns:
+                A ranked (sorted) list containing indices of features  with importance value below importance_thresh
+
+        """
+        # Performing the train_test_split
+        X_train, X_test, y_train, y_test = train_test_split(features, target, test_size=0.25)
```

Comment:
I think here, I might need to assume that the data has been already split into training and test. Therefore, this method is not responsible of the leakage of data from test to train set.

Reply:
Instead, I think I might need to add K-fold cross-validation.